### PR TITLE
OWRank: sort NaNs last; fix sort indicator

### PR DESCRIPTION
**Pull request by kernc**
##### Issue
Fixes https://github.com/biolab/orange3/issues/2617

##### Description of changes
Always sort NaNs last. Implemented by exposing reimplementable `AbstractSortTableModel._argsortData()`. Also fix header indicator by sorting via `view.horizontalHeader().setSortIdicator()` instead of its model.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
This look but.

Would you care to check this error, too?

orange/orange3/Orange/widgets/settings.py:380: UserWarning: Could not read defaults for widget <class 'Orange.widgets.data.owrank.OWRank'>
The following error occurred:

'headerState'
  .format(self.widget_class, ex))

**Comment by ajdapretnar** [comment]
Still some comments.
1) When I sort by, say, Info. gain and chance the data set, the sorting doesn't pertain. I have to re-sort again even though there's an arrow indicating the sorted column.
2) Could we turn off sorting by # or is this not possible with the model?

**Comment by kernc** [comment]
> 1.

Fixed.

> Could we turn off sorting by # or is this not possible with the model?

Sure it's possible. But one may wish to split scores by variable type, or to keep discrete variables with the number of values within a certain threshold. My vote is against.

**Comment by ajdapretnar** [comment]
> But one may wish to split scores by variable type, or to keep discrete variables with the number of values within a certain threshold. My vote is against.

Ok. It is just a little ugly when sorting by #. Perhaps add a tad of extra width then?

screen shot 2017-09-26 at 15 29 58

File: Orange/widgets/data/owrank.py [hunk 1]
```diff
@@ -417,13 +424,14 @@ def updateScores(self):
 
         self.ranksModel.wrap(model_array.tolist())
         self.ranksModel.setHorizontalHeaderLabels(('#',) + labels)
-        self.ranksView.setColumnWidth(0, 30)
+        self.ranksView.setColumnWidth(0, 40)
```
Comment (kernc) [inline Orange/widgets/data/owrank.py]: @ajdapretnar Wide enough?

Comment (ajdapretnar) [reply Orange/widgets/data/owrank.py]: Yes, but now it looks annoyingly wide. 😆 I'm such a brat. I'll merge this as it is a great solution! 👌

**Pull request merged**